sklearn/model_selection/tests/test_validation.py

```diff
@@ -1450,3 +1450,25 @@ def test_fit_and_score():
     # check if the same warning is triggered
     assert_warns_message(FitFailedWarning, warning_message, _fit_and_score,
                          *fit_and_score_args, **fit_and_score_kwargs)
+
+    # check if exception was raised, with default error_score argument
+    assert_raise_message(ValueError, "Failing classifier failed as required",
+                         _fit_and_score, *fit_and_score_args)
+
+    # check if warning was raised, with default error_score argument
+    warning_message = ("Estimator failed to fit for these parameters; the "
+                       "error will be raised. If you want a value to be "
+                       "assigned to the score in such cases, then specify "
+                       "error_score parameter. From version 0.22 the "
+                       "default value will be error_score=np.nan")
+    try:
+        assert_warns_message(FutureWarning, warning_message, _fit_and_score,
+                             *fit_and_score_args)
```

Comment:
Could we use,
```py
with pytest.raises(ValueError):
    assert_warns_message(..)
```
here? As this ``try: except:  pass`` leaves the reader puzzled as to whether it will raise an exception or not.